website/source/api/auth/gcp/index.html.md

```diff
@@ -0,0 +1,362 @@
+---
+layout: "api"
+page_title: "Google Cloud Platform Auth Backend - HTTP API"
+sidebar_current: "docs-http-auth-gcp"
+description: |-
+  This is the API documentation for the Vault GCP authentication
+  backend.
+---
+
+# GCP Auth Backend HTTP API
+
+This is the API documentation for the Vault GCP authentication backend.
+To learn more about the usage and operation, see the
+[Vault GCP backend documentation](/docs/auth/gcp.html).
+
+This documentation assumes the backend is mounted at the `/auth/gcp`
+path in Vault. Since it is possible to mount auth backends at any location,
+please update your API calls accordingly.
+
+## Configure Client
+
+Configures the backend credentials required to perform API calls to GCP.
+These credentials will be used to query the status of IAM entities and get
+service account or other Google public certificates to confirm signed JWTs
+passed in during login.
+
+| Method   | Path                         | Produces               |
+| :------- | :--------------------------- | :--------------------- |
+| `POST`   | `/auth/gcp/config/`          | `204 (empty body)`     |
+
+### Parameters
+
+- `credentials` `(string: "")` - A marshaled JSON string that is the content
+  of a GCP credentials file. If not provided, the Vault server attempts
+  to use [Application Default Credentials](https://developers.google.com/identity/protocols/application-default-credentials)
+
+
+### Sample Payload
+
+```json
+{
+  "credentials": "{ \"type\": \"service_account\", \"project_id\": \"project-123456\",...}"
```

Comment:
technically its a json-marshalled string, so I don't want it to get confused with a regular json object